File: python/examples/enabled_classes.py
# Example script to illustrate how to make API calls to the Private AI API
# to deidentify text using the enabled classes feature.

import dotenv
import os

from privateai_client import PAIClient, request_objects

# Use to load the API KEY for authentication
dotenv.load_dotenv()

# Check if the API_KEY and PAI_URL environment variables are set
if "API_KEY" not in os.environ:
    raise KeyError("API_KEY must be defined in .env to run the examples.")
if "PAI_URL" not in os.environ:
    raise KeyError("PAI_URL must be defined in .env to run the examples.")

# Client initialization
client = PAIClient(url=os.environ["PAI_URL"], api_key=os.environ["API_KEY"])

sample_text = "My name is John and my friend is Grace and we live in Barcelona."
# Create the nested request objects
sample_entity_type_selector = request_objects.entity_type_selector_obj(
    type="ENABLE", value=["NAME", "LOCATION"])
sample_entity_detection = request_objects.entity_detection_obj(
    entity_types=[sample_entity_type_selector])

# Create the request object
process_text_request = request_objects.process_text_obj(
    text=[sample_text], entity_detection=sample_entity_detection)

response = client.process_text(process_text_request)
print(response.processed_text)

File: js/examples/enabled_classes.js
/**
 * Example script to illustrate how to make requests to the Private AI API
 * to deidentify text using the enabled classes feature.
 */
const axios = require("axios");
const dotenv = require("dotenv");

// Use to load the API_KEY and URL
dotenv.config();

// Example without async/await
function sync_enabled_classes() {
  console.log("***** Sync enabled classes *****");
  axios.post(
    `${process.env.PAI_URL}/process/text`,
    {
      text: ["My name is John and my friend is Grace and we live in Barcelona"],
      link_batch: false,
      entity_detection: {
        return_entity: true,
        entity_types: [
          {
            type: "ENABLE",
            value: ["AGE", "LOCATION"],
          },
        ],
      },
      processed_text: {
        type: "MARKER",
        pattern: "[UNIQUE_NUMBERED_ENTITY_TYPE]",
      },
    },
    {
      headers: {
        "Content-Type": "application/json",
        "X-API-KEY": process.env.API_KEY,
      },
    }
  )
  .then((result) => console.log(JSON.stringify(result.data, undefined, 2)))
  .catch((error) =>
    console.error(
      `The request failed with the status code ${error.response.status}`
    )
  );
}

// Example with async/await
async function async_enabled_classes() {
  console.log("***** Async enabled classes *****");
  try {
    const result = await axios.post(
      `${process.env.PAI_URL}/process/text`,
      {
        text: [
          "My name is John and my friend is Grace and we live in Barcelona",
        ],
        link_batch: false,
        entity_detection: {
          return_entity: true,
          entity_types: [
            {
              type: "ENABLE",
              value: ["AGE", "LOCATION"],
            },
          ],
        },
        processed_text: {
          type: "MARKER",
          pattern: "[UNIQUE_NUMBERED_ENTITY_TYPE]",
        },
      },
      {
        headers: {
          "Content-Type": "application/json",
          "X-API-KEY": process.env.API_KEY,
        },
      }
    );

    const { data } = result;

    console.log(JSON.stringify(data, undefined, 2));
  } catch (error) {
    console.error(
      `The request failed with the status code ${error.response.status}`
    );
  }
};

// sync_enabled_classes();
async_enabled_classes();
